sdk/search/Azure.Search.Documents/tests/Batching/BatchingTests.cs

```diff
@@ -0,0 +1,985 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.Search.Documents.Indexes;
+using Azure.Search.Documents.Models;
+using NUnit.Framework;
+
+namespace Azure.Search.Documents.Tests
+{
+    public class BatchingTests : SearchTestBase
+    {
+        public BatchingTests(bool async, SearchClientOptions.ServiceVersion serviceVersion)
+            : base(async, serviceVersion, null /* RecordedTestMode.Record /* to re-record */)
+        {
+        }
+
+        #region Utilities
+        private const int BatchSize = SearchIndexingBufferedSenderOptions<object>.DefaultBatchActionSize;
+        private readonly TimeSpan EventDelay = TimeSpan.FromMilliseconds(250);
+
+        public class SimpleDocument
+        {
+            [SearchableField(IsKey = true)]
```

Comment:
Nit: same comment here about using `SimpleField`.

Reply:
Fixed

Reply:
This was in a few places. Not a big deal, but if people look at this, guiding them to use a simple field would be cheaper for them (index size, that is).